### PR TITLE
Added 'reduceSpectra' function

### PR DESCRIPTION
**Pull request by AharoniLab**
Please evaluate the preliminary version for deisotoping spectra: #282

### CODE REVIEW

**Comment by jorainer** [comment]
This PR adds the functions discussed in issues #282 and #283

File: R/Spectra-functions.R [hunk 1]
```diff
@@ -794,3 +794,25 @@ chunkapply <- function(x, FUN, ..., chunkSize = 1000L, chunks = factor()) {
...
+#' @rdname Spectra
+#'
+#' @author Nir Shahaf, Johannes Rainer
+#'
+#' @export
+deisotopeSpectra <-
+    function(x, substDefinition = isotopicSubstitutionMatrix("HMDB_NEUTRAL"),
+             tolerance = 0, ppm = 10, charge = 1) {
+        im <- force(substDefinition)
+    addProcessing(x, .peaks_deisotope, tolerance = tolerance, ppm = ppm,
```
Comment (lgatto) [inline R/Spectra-functions.R:807]: Please indent all lines the same way.

File: R/Spectra.R [hunk 2]
```diff
@@ -552,6 +552,15 @@ NULL
 #'   the comparison of `x[2]` with `y[3]`). If `SIMPLIFY = TRUE` the `matrix`
 #'   is *simplified* to a `numeric` if length of `x` or `y` is one.
 #'
+#' - `deisotopeSpectra`: *deisotope* each spectrum keeping only the monoisotopic
+#'   peak for groups of isotopologues. Isotopologues are estimated using the
+#'   [isotopologues()] function from the *MetaboCoreUtils* package. Note that
+#'   the default parameters for isotope prediction/detection have been
+#'   determined using data from the Human Metabolome Database (HMDB) and
```
Comment (lgatto) [inline R/Spectra.R:559]: May be add a reference or link to provide more information about *the default parameters for isotope prediction/detection have been determined using data from HMDB*.

File: R/peaks-functions.R [hunk 3]
```diff
@@ -514,3 +514,45 @@ joinPeaksGnps <- function(x, y, xPrecursorMz = NA_real_,
...
     x[keep, , drop = FALSE]
 }
+
+#' Function to keep only the monoisotopic peak for groups of (potential)
+#' isotopologues. The peak with the lowest m/z is considered the monoisotopic
+#' peak for each group of isotopologues.
+#'
+#' @importFrom MetaboCoreUtils isotopologues isotopicSubstitutionMatrix
+#'
+#' @author Nir Shahaf, Johannes Rainer
```
Comment (lgatto) [inline R/peaks-functions.R:524]: May be document the `x` param here with the following (taken from the isotopologues documentation)
```
@param x `matrix` with spectrum data (columns `mz` and `intensity`).
```
and point to `MetaboCoreUtils::isotopologues()` for more details.

File: R/peaks-functions.R [hunk 4]
```diff
@@ -514,3 +514,45 @@ joinPeaksGnps <- function(x, y, xPrecursorMz = NA_real_,
...
+
+#' Function to *reduce* spectra keeping for each group of peaks with highly
+#' similar m/z values (based on provided `tolerance` and `mz` the one with
+#' the highest intensity. The *groups* of mass peaks are defined using the
+#' `MsCoreUtils::group` function.
+#'
+#' @author Nir Shahaf, Johannes Rainer
+#'
+#' @noRd
+.peaks_reduce <- function(x, tolerance = 0, ppm = 10, ...) {
```
Comment (lgatto) [inline R/peaks-functions.R:547]: As above, I would suggest to document `x`.

**Review by lgatto — CHANGES_REQUESTED** [review]
Could you also please update the NEWS file accordingly.

**Comment by jorainer** [comment]
@lgatto all points have been addressed now. Thanks for the review.

**Review by lgatto — APPROVED** [review]

**Comment by lgatto** [comment]
Thank you for the contribution!

**Pull request merged**